packages/styled/flow-tests/base.js

```diff
@@ -1,23 +1,17 @@
 /* eslint-disable no-unused-vars */
 // @flow
-import * as React from 'react'
+import React from 'react'
 import createStyled from '../src/base'
 import type { CreateStyledComponent, StyledComponent } from '../src/utils'
 
-export const valid: CreateStyledComponent = createStyled('div')
+const valid: StyledComponent<empty> = createStyled('div')({
+  color: 'red'
+})
 
 // $FlowExpectError: we can't cast a StyledComponent to string
-export const invalid: string = createStyled('div')
+const invalid: string = createStyled('div')({ color: 'red' })
 
 const styled = createStyled('div')
-type Props = { color: string }
-// prettier-ignore
-const Div = styled<Props>({ color: props => props.color })
```

Comment:
Again, no idea how it used to work... `createStyled()` doesn't return a polymorphic type, only `createStyled.*` are polymorphic types.